client/client.go

```diff
@@ -7,13 +7,13 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 )
 
-// ClientFactoryInterface interface for the client factory
-type ClientFactoryInterface interface {
-	New(url string) ClientInterface
+// IClientFactory interface for the client factory
+type IClientFactory interface {
```

Comment:
Don't you think having this interface is a little bit overkill?

Reply:
There is no other way to mock the rpc client, because, if you go to `func (bs *BatchSynchronizer) resolveWithMember` function, you can see that it is opening a new client based on the committee member url.

It is not a super clean approach I agree, but in the face of non existing abstraction, I think it is good enough.